micro-rdk/src/common/encoder.rs

```diff
@@ -218,8 +243,8 @@ impl Default for FakeEncoder {
 impl FakeEncoder {
     pub fn new() -> Self {
         Self {
-            angle_degrees: 0.0,
-            ticks_per_rotation: 1,
+            angle_degrees: 360.0,
+            ticks: AtomicU32::new(0),
```

Comment:
Why does this need to be an atomic now?

Reply:
it's how we can increment the ticks from inside `get_position` without a mutable reference.
RDK uses a background thread that uses speed and time durations to calculate these, but we don't have that.